modules/gui/warp-list.lua

```diff
@@ -647,13 +850,18 @@ local function role_update_event(event)
     local player = game.players[event.player_index]
     local container = Gui.get_left_element(player, warp_list_container).container
 
+    -- Check if user has permission to add warps
+    local allow_add_warp = check_player_permissions(player, 'allow_add_warp')
+    -- Update container size depending on if the player is allowed to add warps
```

Comment:
```suggestion
    -- Update container size depending on whether the player is allowed to add warps
```
This grammar error seems to be in quite a few places so I'll let you decide whether :wink: you fancy fixing it